int64[] lx
int64[] mx
int64[] rx
int64[] ly
int64[] my
int64[] ry
